onset_maproom/maproom.py

```diff
@@ -91,6 +98,92 @@ def initialize(toto):
     return center_of_the_map, lat_min, lat_max, lon_min, lon_max, lat_label, lon_label
 
 
+def adm_borders(shapes):
+    dbpool = DBPOOL
+    with dbpool.take() as cm:
+        conn = cm.resource
+        with conn:  # transaction
+            s = sql.Composed(
+                [
+                    sql.SQL("with g as ("),
+                    sql.SQL(shapes),
+                    sql.SQL(
+                        """
+                        )
+                        select
+                            g.label, g.key, g.the_geom
+                        from g
+                        """
+                    ),
+                ]
+            )
+            df = pd.read_sql(
+                s,
+                conn,
+            )
+    df["the_geom"] = df["the_geom"].apply(lambda x: wkb.loads(x.tobytes()))
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    return {"features": shapes}
+
+
+@APP.callback(
+    Output("borders_adm1", "data"),
+    Input("submitLatLng","n_clicks"),
+)
+def adm1_borders(toto):
+    return adm_borders(CONFIG["shapes_adm1"])
```

Comment:
I will apply the changes in subseas_... once you approve them, with changes if need be

Reply:
OK. In the meantime, let's use ``Input("location", "pathname")` instead of `submitLatLng` for the dummy input. The Location path never changes after the initial page load, so that will prevent this callback from being called unnecessarily.

Reply:
ok... on it now